File: livingrimoire start here/LivinGrimoire java/LivinGrimoire/GrimoireMemento.java
package LivinGrimoire;

import java.util.Hashtable;

public class GrimoireMemento {
    private AbsDictionaryDB absDictionaryDB;

    public GrimoireMemento(AbsDictionaryDB absDictionaryDB) {
        super();
        this.absDictionaryDB = absDictionaryDB;
    }
    public String simpleLoad(String key){
        return this.absDictionaryDB.load(key);
    }
    public void simpleSave(String key, String value){
        if(key.startsWith("AP")||key.isEmpty()||value.isEmpty()){return;}
        this.absDictionaryDB.save(key,value);
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/livinGrimoire/GrimoireMemento.kt
package livinGrimoire

class GrimoireMemento(private val absDictionaryDB: AbsDictionaryDB) {
    fun simpleLoad(key: String): String {
        return absDictionaryDB.load(key)
    }

    fun simpleSave(key: String, value: String) {
        if (key.isEmpty() || value.isEmpty()) {
            return
        }
        absDictionaryDB.save(key, value)
    }
}